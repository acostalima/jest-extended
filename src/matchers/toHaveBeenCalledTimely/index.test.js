import matcher from './';

expect.extend(matcher);

describe('.toHaveBeenCalledTimely', () => {
  test('fails when mock has not been called', () => {
    const mock = jest.fn();

    expect(() => expect(mock).toHaveBeenCalledTimely([2000])).toThrowErrorMatchingSnapshot();
  });

  test('fails when mock has not been called the mininum required number of times', () => {
    const mock = jest.fn();
    mock();
    mock();
    mock();
    mock.mock.invocationCallOrder[0] = 0;
    mock.mock.invocationCallOrder[1] = 2000;
    mock.mock.invocationCallOrder[2] = 4000;

    expect(() => expect(mock).toHaveBeenCalledTimely([2000, 4000, 16000])).toThrowErrorMatchingSnapshot();
  });

  test('passes when mock is called when mininum specified time intervals have elapsed', () => {
    const mock = jest.fn();
    mock();
    mock();
    mock();
    mock.mock.invocationCallOrder[0] = 0;
    mock.mock.invocationCallOrder[1] = 2000;
    mock.mock.invocationCallOrder[2] = 4000;
    mock.mock.invocationCallOrder[3] = 16000;

    expect(mock).toHaveBeenCalledTimely([2000, 4000, 16000]);
  });
});
