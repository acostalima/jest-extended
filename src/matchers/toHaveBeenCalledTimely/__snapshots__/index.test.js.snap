// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.toHaveBeenCalledTimely fails when mock has not been called 1`] = `
"<dim>expect(</><red>received</><dim>).toHaveBeenCalledTimely(</><green>expected</><dim>)</>

Expected mock to have been called in specified time intervals:
  <green>[2000]</>
Received time intervals:
  <green>[]</>"
`;

exports[`.toHaveBeenCalledTimely fails when mock has not been called the mininum number of times 1`] = `
"<dim>expect(</><red>received</><dim>).toHaveBeenCalledTimely(</><green>expected</><dim>)</>

Expected mock to have been called in specified time intervals:
  <green>[2000, 4000, 16000]</>
Received time intervals:
  <green>[0, 2000, 4000]</>"
`;
